daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -702,10 +706,22 @@ class ParseResult(dp: DataProcessor, override val resultState: PState)
    */
   def validateResult(bytes: Array[Byte]): Unit = {
     Assert.usage(resultState.processorStatus eq Success)
-    val schemaURIStrings = resultState.infoset.asInstanceOf[InfosetElement].runtimeData.schemaURIStringsForFullValidation
+
+    val (v, args) = dp.validationMode match {
+      case ValidationMode.Custom(name, args) =>
+        Validators.find(name).getOrElse(
+          Assert.abort(s"Validator '$name' not found")
```

Comment:
`ValidatorNotFoundException` sounds good

Reply:
Can we move much of this find logic to when a user calls withValidationMode? That way we can detect if a validator exists and has the right args before we try to actually parse things? It would be nice if all the logic here was just the below ``v.validateXML(...)``.

This feels like perhaps it might lead nicely into [DAFFODIL-1749](https://issues.apache.org/jira/browse/DAFFODIL-1749). Or perhaps that should just be done separately, since that's already how our logic behaves right now. Just something to think about as we add this new API to make sure that is possible.

Reply:
I wanted to fail faster then this point, which is done in the CLI, but when using the API you can fall through to here and have to catch the Validator doesnt exist.

There is at least one way to enforce something in the API, but Im not sure its cleaner than doing nothing and just throwing late (eg. factory to produce the CustomValidationMode that ensures a validator exists).

Reply:
> Can we move much of this find logic to when a user calls withValidationMode?

@stevedlawrence I missed your point there earlier.  Yeah that sounds reasonable, Ill move it and see how it looks.

Reply:
@stevedlawrence added the guard in `withValidationMode`, seems clean enough

@mbeckerle 

>I don't think Assert.abort is appropriate here.

I recall now that the assert was there because I was hoping to guard against missing validators making it to that point.  It is changed it over to an exception now, but after the `withValidationMode` change I have the feeling it could still be an assert of some kind.  Im good either way

Reply:
resolved with compilation